lib/Fhp/Model/TanRequestChallengeFlicker/TanRequestChallengeFlicker.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace Fhp\Model\TanRequestChallengeFlicker;
+
+use Fhp\Syntax\Bin;
+use InvalidArgumentException;
+
+class TanRequestChallengeFlicker
+{
+    /**
+     * @var string original challenge data
+     */
+    private $challenge;
+
+    /**
+     * @var FlickerTanStartCode holds and parses the startcode block of the challenge
+     */
+    private $startCode;
+
+    /**
+     * @var FlickerTanDataElement Holds and parses the first DataElement of the challenge
+     */
+    private $de1;
```

Comment:
These three are never treated differently, so how about making them an array? It would reduce code duplication in the places where they're used.

Reply:
Also note this part of the spec: "Leere Datenelemente am Ende des Datenstromes können komplett inklusive Längenfeld entfallen."

Reply:
I did it the way i did because 3 is the maximum. But ofc an array would work as well. Will have a look into it

Reply:
done